compiler/daml-lf-verify/src/DA/Daml/LF/Verify/Generate.hs

```diff
@@ -141,9 +141,16 @@ genChoice pac tem (this',this) temFs TemplateChoice{..} = do
   extVarEnv self
   extVarEnv arg
   argFs <- recTypFields (snd chcArgBinder)
+  -- Replace the argument and template fields with their record projection form.
+  let argFieldSubst = foldl concatExprSubst emptyExprSubst
+        (map (\f -> singleExprSubst (fieldName2VarName f) (EStructProj f (EVar arg))) argFs)
+      thisFieldSubst = foldl concatExprSubst emptyExprSubst
+        (map (\f -> singleExprSubst (fieldName2VarName f) (EStructProj f (EVar this))) temFs)
```

Comment:
Here and for `argFieldSubst` above:
```suggestion
      thisFieldSubst = Map.fromList [(fieldName2VarName f, EStructProj f (EVar this)) | f <- temFs]
```
You probably want to give `Map.fromList` a name like `substFromList` in `Subst.hs`.